apps/api/src/report/report.service.ts

```diff
@@ -221,37 +221,149 @@ export class ReportService {
    */
   async splitReportFourNewOldProcess(f: string, t: string) {
     const statuses = await this.getStatusMap();
-    const entityManager = getManager();
-
     const { from, to } = this.captureFromTo(f, t);
 
     this.logger.log(`getLicensingStageApplicants: Apply date filter from (${from}) and to (${to})`);
+    const connection = getConnection();
+    const mappedStatuses = LIC_REG_STAGE.map(licensingStatus => {
+      return statuses[licensingStatus];
+    });
+    const mappedStatusesString = mappedStatuses.map(status => `'${status}'`).join(',');
+    try {
+      const oldProcess = await connection.query(
+        this.reportUtilService.reportFour(mappedStatusesString, from, to, false),
+      );
+      const newProcess = await connection.query(
+        this.reportUtilService.reportFour(mappedStatusesString, from, to, true),
+      );
+      const licenseResults = await this.countLicense(connection);
+      const withdrawnOld =
+        (await connection.query(this.reportUtilService.getWithdrawn(to, from, false)))[0].count ||
+        '0';
+      const withdrawnNew =
+        (await connection.query(this.reportUtilService.getWithdrawn(to, from, true)))[0].count ||
+        '0';
+      return this.mapReportFourResults(
+        statuses,
+        oldProcess,
+        newProcess,
+        licenseResults,
+        withdrawnOld,
+        withdrawnNew,
+      );
+    } catch (e) {
+      this.logger.error(e);
+      return [];
+    }
+  }
+  mapReportFourResults(
+    statuses: Record<string, string>,
+    oldProcess: { status_id: string; count: string }[],
+    newProcess: { status_id: string; count: string }[],
+    licenceResults: any,
+    withdrawnOld: string,
+    withdrawnNew: string,
+  ) {
+    let report = REPORT_FOUR_STEPS.map((step: string) => {
+      switch (step) {
+        case 'Granted full licensure':
+          return {
+            status: step,
+            oldProcessApplicants: licenceResults.oldFullLicence[0]?.count || '0',
+            newProcessApplicants: licenceResults.newFullLicence[0]?.count || '0',
+          };
```

Comment:
wonder if it's worth just creating a function to return this object to reduce the repetitiveness